golang/cosmos/x/swingset/keeper/keeper.go

```diff
@@ -117,37 +121,47 @@ func (k Keeper) PushAction(ctx sdk.Context, action vm.Jsonable) error {
 	if !txHashOk || !msgIdxOk {
 		stdlog.Printf("error while extracting context for action %q\n", action)
 	}
-	record := actionQueueRecord{Action: action, Context: actionContext{BlockHeight: ctx.BlockHeight(), TxHash: txHash, MsgIdx: msgIdx}}
+	record := inboundQueueRecord{Action: action, Context: actionContext{BlockHeight: ctx.BlockHeight(), TxHash: txHash, MsgIdx: msgIdx}}
 	bz, err := json.Marshal(record)
 	if err != nil {
 		return err
 	}
 
 	// Get the current queue tail, defaulting to zero if its vstorage doesn't exist.
 	// The `tail` is the value of the next index to be inserted
-	tail, err := k.actionQueueIndex(ctx, "tail")
+	tail, err := k.queueIndex(ctx, inboundQueuePath, "tail")
 	if err != nil {
 		return err
 	}
 
 	if tail.Equal(MaxSDKInt) {
-		return errors.New(StoragePathActionQueue + " overflow")
+		return errors.New(inboundQueuePath + " overflow")
 	}
 	nextTail := tail.Add(sdk.NewInt(1))
 
 	// Set the vstorage corresponding to the queue entry for the current tail.
-	path := StoragePathActionQueue + "." + tail.String()
+	path := inboundQueuePath + "." + tail.String()
 	k.vstorageKeeper.SetStorage(ctx, vstoragetypes.NewStorageEntry(path, string(bz)))
 
 	// Update the tail to point to the next available entry.
-	path = StoragePathActionQueue + ".tail"
+	path = inboundQueuePath + ".tail"
 	k.vstorageKeeper.SetStorage(ctx, vstoragetypes.NewStorageEntry(path, nextTail.String()))
 	return nil
 }
 
-func (k Keeper) actionQueueIndex(ctx sdk.Context, position string) (sdk.Int, error) {
+// PushAction appends an action to the controller's actionQueue.
+func (k Keeper) PushAction(ctx sdk.Context, action vm.Jsonable) error {
+	return k.pushAction(ctx, StoragePathActionQueue, action)
+}
+
+// PushAction appends an action to the controller's highPriorityQueue.
+func (k Keeper) PushHighPriorityAction(ctx sdk.Context, action vm.Jsonable) error {
+	return k.pushAction(ctx, StoragePathHighPriorityQueue, action)
+}
+
+func (k Keeper) queueIndex(ctx sdk.Context, queuePath string, position string) (sdk.Int, error) {
```

Comment:
Suggestion: Move the string concatenation into the caller (where it already exists anyway in the case of `pushAction`) and refactor into something like
```suggestion
func (k Keeper) getInt(ctx sdk.Context, path string, default *sdk.Int) (sdk.Int, error) {
```
```diff
 	// Get the current queue tail, defaulting to zero if its vstorage doesn't exist.
 	// The `tail` is the value of the next index to be inserted
-	tail, err := k.queueIndex(ctx, inboundQueuePath, "tail")
+	tail, err := k.getInt(ctx, inboundQueuePath + ".tail", zero)
 	if err != nil {
 		return err
 	}
```
```diff
 func (k Keeper) queueLength(ctx sdk.Context, queuePath string) (sdk.Int, error) {
-	head, err := k.queueIndex(ctx, queuePath, "head")
+	head, err := k.getInt(ctx, queuePath + ".head", zero)
 	if err != nil {
 		return sdk.NewInt(0), err
 	}
-	tail, err := k.queueIndex(ctx, queuePath, "tail")
+	tail, err := k.getInt(ctx, queuePath + ".tail", zero)
 	if err != nil {
 		return sdk.NewInt(0), err
 	}
 }
```

It could even be moved down into `vstorage/keeper`, and/or the queue concept could be abstracted here into e.g. `queuePush(ctx sdk.Context, queuePath string) error` and `queueLength(ctx sdk.Context, queuePath string) (sdk.Int, error)` rather than the mismatched `pushAction` and `queueLength`.

Reply:
Refactored and moved to vstorage mostly as suggested (minus the default value).